pylib/mozhginfo/tests/test_pushlog_client.py

```diff
@@ -0,0 +1,155 @@
+import json
+import unittest
+
+from mock import patch, Mock
+from mozhginfo import pushlog_client
+from mozhginfo.pushlog_client import (
+    query_revisions_range,
+    query_revision_info,
+    query_full_revision_info,
+    query_pushid_range,
+    query_repo_tip,
+    query_revisions_range_from_revision_before_and_after
+)
+
+LIST_REVISION = """
+{
+    "lastpushid": 67951,
+    "pushes": {
+    "53348":{"changesets": ["eb15e3f893453d6a4472f8905271aba33f8b68d5"],
+    "date": 1419851870, "user": "nobody@mozilla.com"},
+    "53349": {"changesets": ["1c5b4332e2f1b73fe03977b69371e9a08503bff3"],
+    "date": 1419852144, "user": "nobody@gmail.com"},
+    "53350": {"changesets": ["724f0a71d62171da1357e6c1f93453359e54206b"],
+    "date": 1419853433, "user": "nobody@mozilla.com"}
+    }
+}
+"""
+
+REVISION_INFO_REPOSITORIES = """{
+    "87833": {
+    "changesets": ["71e69424094d2f86c51ba544fd861d65a578a0f2"],
+    "date": 1441983569,
+    "user": "nobody@mozilla.com"
+    }
+}
+"""
+
+INVALID_REVISION = """
+"unknown revision '123456123456s'"
+"""
+
+GOOD_REVISION = """
+{
+ "82366": {
+  "changesets": [
+   "4e030c8cf8c35158c9924f6bb33ffe8af00c162b"
+  ],
+  "date": 1438992451,
+  "user": "nobody@mozilla.com"
+ }
+}
+"""
+
+
+def mock_response(content, status):
+    """Mock of requests.get()."""
+    response = Mock()
+    response.content = content
+
+    def mock_response_json():
+        return json.loads(content)
+
+    response.json = mock_response_json
+    response.status_code = status
+    response.reason = 'OK'
+    return response
+
+
+class TestQueries(unittest.TestCase):
+
+    def setUp(self):
+        self.repo_url = "https://hg.mozilla.org/integration/mozilla-inbound"
+        self.revision = '71e69424094d'
+        self.start_revision = '8850aa0f93453d6a4472f8905271aba33f8b68d5'
+        self.end_revision = '2a193b7fd62171da1357e6c1f93453359e54206b'
+
+    @patch('requests.get', return_value=mock_response(LIST_REVISION, 200))
```

Comment:
We try not to use mocks in version-control-tools because they get out of sync with reality. Instead, our tests start actual Mercurial servers configured like hg.mozilla.org and tests hit them. See e.g. https://hg.mozilla.org/hgcustom/version-control-tools/file/tip/hgext/pushlog-legacy/tests/test-hgweb.t. https://www.mercurial-scm.org/wiki/WritingTests describes the test format. You can even write inline Python doctest-style tests. So you can e.g. start a server in the shell then have some boilerplate python query the server.

Anyway, we'll probably want to rewrite these tests at some time. This isn't a blocker for landing, however.